### ISSUE TITLE
Double click line event 

### ISSUE THREAD

**Issue by mrunks** [Blazor-Diagrams/Blazor.Diagrams#124]
Looking at the custom link examples, I was trying to find a way that I could implement an event to remove a link by selecting the line either double click or right click.

I tried adding it to the <line> element such as  @ondblclick=(() => Remove()), but was unable to get a reference to the parent Node to remove the link.  

Is there an existing example or way to remove a link by clicking, double clicking or right clicking on the line that one wishes to remove ?

**Comment by zHaytam** [comment]
Hello,
Here's a simple solution for the Left/Right click:

```cs
// Assuming your diagram instance is called Diagram,
// After you initialize it, do:
Diagram.MouseClick += OnMouseClick;

private void OnMouseClick(Model model, MouseEventArgs e)
{
    if (model is BaseLinkModel link)
    {
        Diagram.Links.Remove(link);
    }
}
```

This is untested, by it should work, or at least give you the idea.

For the double click, there isn't an event for that YET, but you can simply do it by timing the MouseClick event, if two events were registered quickly, then you can count it as a double click.

Hope this helps!

**Comment by mrunks** [comment]
Since the actions for individual components come for the diagram events as opposed to other components i assume the if one wanted a double click on a line and another double click for a node etc then there would be different "if" checks inside the mouseClick onthe diagram.  

Right click would be important to have as well at least for what I do. 

As mentioned earlier I just am going through are existing app which uses JS Plumb and trying replicate the features in the blazor equivalent.  I really like what you have done great work!

**Comment by zHaytam** [comment]
Right, you'll need to check the type of model and its information. We try our best to keep UI things from the models, that separation is important. That's why UI events are available in the diagram, not the models.

For the right click, just check the `MouseEventArgs`'s `Button` property if it equals 2.

**Comment by zHaytam** [comment]
This is now available in 2.1.4!
